### PR TITLE
Add kick participant and end meeting actions for meeting hosts

### PR DESCRIPTION
**Pull request by richtrwhereby**
### Description

Allow participants joining with the `host` role to kick a remote participant and/or end the meeting for all remote participants.

### Testing

1. Run local-stack and join a room with 1 or more participants via PWA.
2. Check out this branch
3. Ensure that STORYBOOK_ROOM is the same URL as the joined room in step 1.
4. In the `/.env` file add a valid _host_ `roomKey` for the configured `STORYBOOK_ROOM` in following line:

```
STORYBOOK_ROOM_HOST_ROOMKEY=[host_roomKey_here]
```

5. Run `yarn build && yarn dev`
6. Open the storybook story: "Custom UI" > "Room Connection With Host Controls" ([direct localhost link](http://localhost:6006/?path=/story/examples-custom-ui--room-connection-with-host-controls))
7. Check that the `roomKey` is filled with the host room key added in step 4.
8. Clicking on the "End meeting" button and individual "Kick" buttons should now perform the respective actions in the connected room [Screenshot 1].
9. In the storybook controls panel, remove the `roomKey` parameter and reload the story
10. The "End meeting" button and individual "Kick" buttons shown in the UI should now be in red [Screenshot 2]
11. Clicking on the "End meeting" button or any of the individual "Kick" buttons should not perform the action in the connected room (not connected as a "host" now).
12. Additionally, a warning should be displayed in the console whenever these actions are invoked by a non-"host" user.

#### Screenshot 1: "host" `roomKey` provided

Screenshot 2024-04-17 at 15 47 04

#### Screenshot 2: "host" `roomKey` not provided

Screenshot 2024-04-17 at 15 47 17

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [x] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

File: packages/core/src/redux/slices/room.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+export const doEndMeeting = createAppAuthorizedThunk(
+    (state) => selectIsAuthorizedToEndMeeting(state),
+    () => (_, getState) => {
+        const state = getState();
+
+        const clientsToKick = selectRemoteParticipants(state).map((c) => c.id);
+
+        const { socket } = selectSignalConnectionRaw(state);
+
+        socket?.emit("kick_client", { clientIds: clientsToKick, reasonId: "end-meeting" });
```
Comment (thyal) [inline packages/core/src/redux/slices/room.ts:106]:
```suggestion
     if(clientsToKick.length) {
        const { socket } = selectSignalConnectionRaw(state);

        socket?.emit("kick_client", { clientIds: clientsToKick, reasonId: "end-meeting" });
      }
```
Let's wrap this in a guard, and not call signal unless we have some clients to actually kick. I got disconnected from the room and the kick event had an error message when I clicked this and I was alone in the room

**Review by thyal — COMMENTED** [review]
Overall looks good, but I think we should move the `roomKey` back to the authorization slice. Also added a comment about a guard in the `doEndMeeting` action creator.

File: packages/core/src/redux/slices/authorization.ts [hunk 2]
```diff
@@ -1,100 +1,58 @@
...
  */
 
 export interface AuthorizationState {
-    roomKey: string | null;
-    roomLocked: boolean;
+    roleName: RoleName;
 }
 
 const initialState: AuthorizationState = {
-    roomKey: null,
```
[the remark this answers is not in the document]
Comment (richtrwhereby) [reply packages/core/src/redux/slices/authorization.ts:25]: Agree and put `roomKey` handling back in to the authorization slice in 8115acca5e8221519b56da898c327f6508a0f1c5.

File: packages/core/src/redux/slices/room.ts (hunk 1 above)
Comment (richtrwhereby) [reply packages/core/src/redux/slices/room.ts]: Great catch and added in 8115acca5e8221519b56da898c327f6508a0f1c5.

**Review by thyal — APPROVED** [review]
🍰 Nice!

**Pull request merged**